### ISSUE TITLE
Thin backend layer between Extractor and NewPipe client?

### ISSUE THREAD

**Issue by Bluesir9** [TeamNewPipe/NewPipeExtractor#154]
Maybe the approach has already been considered in the past but since I couldn't find an issue about the same, I decided to make a fresh one. 

Given we have no control over how frequently and/or drastically YouTube changes the way it serves its pages, maybe its worth considering writing a thin layer sitting between the extractor and the client? That way if and when the extractor starts failing due to changes made on YouTube's end, the client can still continue to consume a subset of "cached" content on the thin layer. 

Not sure if there are any legal or miscellaneous blockades that stop us from doing this, but thought it was best to clarify by simply asking :)

**Comment by theScrabi** [comment]
You mean hosted? Hosted might get us into legal trouble.

**Comment by Bluesir9** [comment]
No, the backend will not host any videos. It will only cache search results so that if the Extractor is not able to get the latest videos from YouTube for whatever reason, we can continue serving these cached results and keep the app operational.
